public/app/features/dashboard/dashboard_migration.ts

```diff
@@ -363,6 +363,11 @@ export class DashboardMigrator {
 
     if (oldVersion < 16) {
       this.upgradeToGridLayout(old);
+      panelUpgrades.push(function(panel) {
```

Comment:
should this not be inside the upgradeToGridLayout ?

Reply:
need to pass panelUpgrades to upgradeToGridLayout(), is it okay?

Reply:
I first write there, but test fails.
I can't understand why it fails, and then I think the place is not correct, I should use panelUpgrades.